app/controllers/homes_controller.rb

```diff
@@ -1,5 +1,11 @@
 class HomesController < ApplicationController
   def index
+    @products = Product.all
+  end
+
+  def profile
+    @user = User.find(params[:id])
+    @products=Product.where("user_id=?", current_user.id)
```

Comment:
Instead of using where, we can use join. As I guess, the current user and the id in the params are the same